ui/src/main/java/ucar/nc2/ui/simplegeom/SimpleGeomTable.java

```diff
@@ -0,0 +1,859 @@
+    for (GridDatatype g : list2)
+        sgList.add(new SimpleGeomBean(g));
+      varTable.setBeans(sgList);
+    
+
+    if (csTable != null) {
+      List<GeoCoordinateSystemBean> csList = new ArrayList<GeoCoordinateSystemBean>();
+      List<GeoAxisBean> axisList;
+      axisList = new ArrayList<GeoAxisBean>();
+      for (GridDataset.Gridset gset : gridDataset.getGridsets()) {
+        csList.add(new GeoCoordinateSystemBean(gset));
+        GridCoordSystem gsys = gset.getGeoCoordSystem();
+        List<CoordinateAxis> axes = gsys.getCoordinateAxes();
+        for (int i = 0; i < axes.size(); i++) {
+          CoordinateAxis axis = axes.get(i);
+          GeoAxisBean axisBean = new GeoAxisBean(axis);
+          if (!contains(axisList, axisBean.getName()))
+            axisList.add(axisBean);
+        }
+      }
+      csTable.setBeans(csList);
+      axisTable.setBeans(axisList);
+    }
+  }
+
+  public void setDataset(GridDataset gds) throws IOException {
+    this.gridDataset = gds;
+
+    List<GeoGridBean> beanList = new ArrayList<GeoGridBean>();
+    java.util.List<GridDatatype> list = gridDataset.getGrids();
+    for (GridDatatype g : list)
+      beanList.add(new GeoGridBean(g));
+    varTable.setBeans(beanList);
+
+    if (csTable != null) {
+      List<GeoCoordinateSystemBean> csList = new ArrayList<GeoCoordinateSystemBean>();
+      List<GeoAxisBean> axisList;
+      axisList = new ArrayList<GeoAxisBean>();
+      for (GridDataset.Gridset gset : gridDataset.getGridsets()) {
+        csList.add(new GeoCoordinateSystemBean(gset));
+        GridCoordSystem gsys = gset.getGeoCoordSystem();
+        List<CoordinateAxis> axes = gsys.getCoordinateAxes();
+        for (int i = 0; i < axes.size(); i++) {
+          CoordinateAxis axis = axes.get(i);
+          GeoAxisBean axisBean = new GeoAxisBean(axis);
+          if (!contains(axisList, axisBean.getName()))
+            axisList.add(axisBean);
+        }
+      }
+      csTable.setBeans(csList);
+      axisTable.setBeans(axisList);
+    }
+  }
+
+  private boolean contains(List<GeoAxisBean> axisList, String name) {
+    for (GeoAxisBean axis : axisList)
+      if (axis.getName().equals(name)) return true;
+    return false;
+  }
+
+  public GridDataset getGridDataset() {
+    return gridDataset;
+  }
+
+  public List<String> getSelectedGrids() {
+    List grids = varTable.getSelectedBeans();
+    List<String> result = new ArrayList<String>();
+    for (Object bean : grids) {
+      GeoGridBean gbean = (GeoGridBean) bean;
+      result.add(gbean.getName());
+    }
+    return result;
+  }
+
+
+  public GridDatatype getGrid() {
+    GeoGridBean vb = (GeoGridBean) varTable.getSelectedBean();
+    if (vb == null) {
+      List grids = gridDataset.getGrids();
+      if (grids.size() > 0)
+        return (GridDatatype) grids.get(0);
+      else
+        return null;
+    }
+    return gridDataset.findGridDatatype(vb.getName());
+  }
+
+  public class GeoGridBean {

```

Comment:
Classes tend to be easier to find and reason about when they each get their own file. There are exceptions to this tendency though. Is there a strong case for keeping this class inside of this file? Would it be prohibitively time-consuming to move this class to a standalone file?

Reply:
as discussed in standup2, leaving in for now.